sql/sql_repl.cc

```diff
@@ -2232,15 +2239,118 @@ send_event_to_slave(binlog_send_info *info, Log_event_type event_type,
   }
 
   /*
-    Skip events with the @@skip_replication flag set, if slave requested
-    skipping of such events.
+    Skip events that have either
+      1) @@skip_replication flag
+      2) Set during binlogging (set using binlog_dump_* filter) 
+
   */
-  if (info->thd->variables.option_bits & OPTION_SKIP_REPLICATION)
+  if (info->thd->variables.option_bits & OPTION_SKIP_REPLICATION ||
+    !binlog_dump_filter->is_db_empty() || binlog_dump_filter->is_on())
```

Comment:
I don't think we need to check the binlog_dump_filter at this point in the life-cycle, only
`OPTION_SKIP_REPLICATION` because that should be sorted out at binlog time.

Reply:
Ah I misunderstood what this check was doing. The check for `(info->thd->variables.option_bits & OPTION_SKIP_REPLICATION)` is only the server option, I had it in my head it was _also_ the check for `(event_flags & LOG_EVENT_SKIP_REPLICATION_F)` (i.e. the actual binlog-flagged option). And I also forgot that the `Rpl_filter` API is confusing :( I would think `is_on()` covers both db and table rules, but it doesn't.

So then you are right. It should be ok (I'll file something to clean up the `Rpl_filter` API for another time, I think).